config/management/src/lib.rs

```diff
@@ -404,27 +431,68 @@ pub mod tests {
+        let operator_key = storage_helper
+            .operator_key(local_operator_ns, remote_operator_ns)
+            .unwrap();
+
+        // Upload an owner key to the remote storage
+        let owner_name = "owner";
+        let owner_key = Ed25519PrivateKey::generate_for_testing().public_key();
+        let owner_account = account_address::from_public_key(&owner_key);
+        let mut remote_storage = storage_helper.storage(owner_name.into());
+        remote_storage
+            .set(OWNER_KEY, Value::Ed25519PublicKey(owner_key))
+            .map_err(|e| Error::RemoteStorageWriteError(OWNER_KEY, e.to_string()))
+            .unwrap();
 
-        let local_txn = helper
+        // Operator calls the validator-config command
+        let local_config_tx = storage_helper
             .validator_config(
-                AccountAddress::random(),
+                owner_name,
                 "/ip4/0.0.0.0/tcp/6180".parse().unwrap(),
                 "/ip4/0.0.0.0/tcp/6180".parse().unwrap(),
-                local_ns,
-                remote_ns,
+                local_operator_ns,
+                remote_operator_ns,
             )
             .unwrap();
 
-        let remote = helper.storage(remote_ns.into());
-        let remote_txn = remote.get(constants::VALIDATOR_CONFIG).unwrap().value;
-        let remote_txn = remote_txn.transaction().unwrap();
+        // Verify that a validator config transaction was uploaded to the remote storage
+        let remote_storage = storage_helper.storage(remote_operator_ns.into());
+        let uploaded_config_tx = remote_storage
+            .get(constants::VALIDATOR_CONFIG)
+            .unwrap()
+            .value
+            .transaction()
+            .unwrap();
+        assert_eq!(local_config_tx, uploaded_config_tx);
+
+        // Verify the transaction sender is the operator account address
+        let operator_account = account_address::from_public_key(&operator_key);
+        let uploaded_user_transaction = uploaded_config_tx.as_signed_user_txn().unwrap();
+        assert_eq!(operator_account, uploaded_user_transaction.sender());
+
+        // Verify the validator config in the transaction has the correct account address
+        match uploaded_user_transaction.payload() {
+            TransactionPayload::Script(script) => {
+                assert_eq!(6, script.args().len());
 
-        assert_eq!(local_txn, remote_txn);
+                match script.args().get(0).unwrap() {
+                    TransactionArgument::Address(account_address) => {
+                        assert_eq!(&owner_account, account_address);
+                    }
+                    _ => panic!(
+                        "Found an invalid argument type for the validator-config transaction script!"
+                    ),
+                };
+            }
+            _ => panic!("Invalid validator-config transaction payload found!"),
+        };
```

Comment:
this feels a little heavy considering the other tests we have... the original purpose of this test was just to ensure that the functions CLI interaction remained consistent.

Reply:
Interesting.. I kinda liked being able to assert the outcome of the given CLI command (i.e., ensure that it's doing what we expect on the backend. It definitely made this PR easier to reason about/verify correctness. If you would prefer not to test this, I can just truncate the backend assertions.